src/error.rs

```diff
@@ -0,0 +1,87 @@
+//! Implements Error type. It implements the `std::error::Error` for errors returned by ODBC and
+//! allows for an error handling which his idomatic to rust
+
+use super::raw;
+use std::fmt::{Display, Formatter};
+use std;
+
+/// ODBC Diagnostic Record
+#[derive(Debug)]
+pub struct DiagRec {
+    pub state: [raw::SQLCHAR; 6],
+    pub native_error_pointer: raw::SQLINTEGER,
+    pub message: String,
+}
+
+impl DiagRec {
+    /// Reads first diagnostic record from handle
```

Comment:
I think we should support reading all the diagnostic records - not only the first one. And  I'm not quite sure that we have to panic if there is no diagnostic record (SQL_NO_DATA case)